workflows/host-genome-generation/host_genome_generation.wdl

```diff
@@ -1,150 +1,341 @@
-        # Generate STAR genome
-        #
-
-        # Make directory for STAR genome
-        STAR_GENOME="~{host_name}_STAR_genome"
-        # HACK: we used to support splitting star indexes into many parts, this made things slower
-        # Here we generate the index as if it is in many parts, but there is only ever one part for
-        # backwards compatibility
-        mkdir -p "$STAR_GENOME/part-0"
-
-        AVAILABLE_MEMORY=$(free --bytes | head -n 2 | tail -n 1 | sed "s/  */ /g" | cut -d' ' -f 7)
-
-        STAR \
-          --sjdbGTFfile $GTF_PATH \
-          --runThreadN $(nproc) \
-          --runMode genomeGenerate \
-          --genomeFastaFiles fasta_with_ercc.fa \
-          --limitGenomeGenerateRAM $AVAILABLE_MEMORY \
-          --genomeDir "$STAR_GENOME/part-0"
-
-        # create a parts.txt file for backwards compatibility
-        echo 1 > "$STAR_GENOME/parts.txt"
-
-        # tar STAR genome
-        tar cvf "$STAR_GENOME.tar" -C $(pwd) $STAR_GENOME
-
-        #
-        # Generate bowtie2 genome
-        #
-
-        # Make directory for bowtie2 genome
-        BOWTIE2_GENOME="~{host_name}_bowtie2_genome"
-        mkdir $BOWTIE2_GENOME
-
-        # Change into the directory to contain the output and generate bowtie2 genome
-        cd $BOWTIE2_GENOME
-        bowtie2-build ../fasta_with_ercc.fa "~{host_name}"
-        cd ..
-
-        # tar bowtie2 genome
-        tar cvf "$BOWTIE2_GENOME.tar" -C $(pwd) $BOWTIE2_GENOME
-
-        minimap2 -x map-ont -d "~{host_name}_minimap2_genome_dna.mmi" fasta_with_ercc.fa
-        minimap2 -x splice -d "~{host_name}_minimap2_genome_rna.mmi" fasta_with_ercc.fa
-    >>>
-
-    output {
-        File original_input_fasta = "input.fa"
-        File? original_input_gtf = "input.gtf"
-        File fasta_with_ercc_fa = "fasta_with_ercc.fa"
-        File? gtf_with_ercc_gtf = "gtf_with_ercc.gtf"
-        File star_genome_tar = "~{host_name}_STAR_genome.tar"
-        File bowtie_genome_tar = "~{host_name}_bowtie2_genome.tar"
-        File minimap2_dna = "~{host_name}_minimap2_genome_dna.mmi"
-        File minimap2_rna = "~{host_name}_minimap2_genome_rna.mmi"
-    }
-
-    runtime {
-        docker: docker_image_id
-    }
+task kallisto_index {
+  input {
+    Array[File] transcripts_fasta_gz
+    String genome_name
+
+    String docker_image_id
+  }
+
+  String idx_fn = "~{genome_name}.kallisto.idx"
+  command <<<
+    set -euxo pipefail
+    /kallisto/kallisto index --index '~{idx_fn}' ~{sep(' ',transcripts_fasta_gz)}
+    >&2 ls -l
+  >>>
+
+  output {
+    File idx = idx_fn
+  }
+
+  runtime {
+    docker: docker_image_id
+    memory: "16GiB"
+  }
+}
+
+task minimap2_index {
+  input {
+    File fasta
+    String genome_name
+    String nucleotide_type
+
+    String docker_image_id
+  }
+
+  command <<<
+    set -euxo pipefail
+    TMPDIR=${TMPDIR:-/tmp}
+
+    if [ "~{nucleotide_type}" == "dna" ]; then
+        >&2 minimap2 -x map-ont -d '~{genome_name}_~{nucleotide_type}.mmi' "~{fasta}"
```

Comment:
Not sure why, but `nucleotide_type` is not getting filled in with `dna` or `rna` but `genome_name` is.